wpilibc/src/main/native/include/SampleRobot.h

```diff
@@ -7,11 +7,17 @@
 
 #pragma once
 
+#include <support/deprecated.h>
+
 #include "RobotBase.h"
 
 namespace frc {
 
-class SampleRobot : public RobotBase {
+class WPI_DEPRECATED(
+    "WARNING: While it may look like a good choice to use for your code if "
```

Comment:
Is the word "WARNING" necessary since it's contained within a compiler warning? I guess it depends on whether people heed compiler warnings and whether they heed it any more with the word "WARNING" included.

Reply:
I think that having the all caps word in the source will bring it to people's attention before they start writing code.

Reply:
Will this cause deprecated warnings for all project (even those that don't use Sample Robot) like was recently seen with Robot Drive?

Reply:
The other patch errored on functions that took a RobotDrive instance. There's no function taking a SampleRobot instance, this should be fine. Can we recommend TimedRobot instead of IterativeRobot?